src/Core/Core.js

```diff
@@ -11972,16 +11971,16 @@ export default class Core {
 
           //TODO: need type warning (red but doesn't hang around)
           if (data.viewedSolution) {
-            if (!suppressToast) {
-              postMessage({
-                messageType: "sendToast",
-                coreId: this.coreId,
-                args: {
-                  message: "No credit awarded since solution was viewed.",
-                  toastType: toastType.INFO,
-                },
-              });
-            }
+            // if (!suppressToast) {
```

Comment:
This messageType should be switched to "sendAlert" to get rid of the technical debt and have it link with the listener already in PageViewer. (The code appears to be only half converted from "toast" to "alert", so we should complete the conversion if we are restoring alerts.) In the new repository, I switched to `alertType: "info"`, so we might as well move to that.

Also, we need these alerts for the due date being passed, the time limit being passed, and number of attempts exceeded. (Not sure about the error conditions.)

Reply:
done